tests/reframe/eessi-checks/applications/gromacs.py

```diff
@@ -0,0 +1,78 @@
+import re
+import reframe as rfm
+from reframe.utility import find_modules
+
+from testlib.applications.gromacs import Gromacs
+
+@rfm.required_version('>=3.5.0')
+@rfm.simple_test
+class Gromacs_EESSI(Gromacs):
+    '''EESSI Gromacs check.
+    This test will run GROMACS using all modules with 'GROMACS' in the module environment it can find.
+    On GPU nodes, it will only run tests if module names also contain 'cuda'.
+    On CPU nodes, it will only run tests if a module name does NOT contain 'cuda'.
+    Whether a nodes is CPU/GPU is determined based on if a device named 'gpu' is specified in the ReFrame settings file for the current partition.
+    Number of tasks, tasks per node and cpus per task are set based on the number of GPUs and number of CPUs specified in the ReFrame config file for the current partition. 
+    '''
+
+    scale = parameter(['singlenode', 'small', 'large'])
+    module_info = parameter(find_modules('GROMACS', environ_mapping={r'.*': 'builtin'}))
+
+    @rfm.run_after('init')
+    def apply_module_info(self):
+        self.s, self.e, self.m = self.module_info
+        self.valid_systems = [self.s]
+        self.modules = [self.m]
+        self.valid_prog_environs = [self.e]
+
+    @rfm.run_after('init')
+    def set_test_scale(self):
```

Comment:
This comment is just a note on the style for the test parameterisation :)
If you're not too bothered about how the name of the test gets printed out, I would write the parameter `scale` as
```python
    scale = parameter([
        ('singlenode', 10000, 1),
        ('small', 40000, 4),
        ('large', 100000, 10),
    ])
```
so then this hook just becomes
```python
@run_after('init')
def set_test_scale(self):
    scale_variant, self.nsteps, self.num_nodes = self.scale
    self.tags.add(scale_variant)
```
In any case, the naming of the test with long parameters should be fixed in the future, since there is an open PR on that.

Reply:
Agreed, much cleaner. I'm not to worried about the test names: they are a bit messy anyway if you use the automatic module discovery tbh... :)